### DISCUSSION TITLE
How can I use this trigger without leaking my service_role_key

### DISCUSSION THREAD

**Discussion by BragdonD** [supabase/supabase#19573] [Questions]
Hey,

I am following the tutorial to create an app where user can create an account and update profile picture. It is the one provide inside the documentation [here](https://supabase.com/docs/guides/getting-started/tutorials/with-expo-react-native?auth-store=async-storage&database-method=sql).

Inside the documentation, the ``url`` and ``service role key`` are written directly inside the sql code, which is leaking the service role key inside the git project which is not recommended by the documentation [here](https://supabase.com/docs/guides/api/api-keys).

So how am I suppose to use the following snippet of code:

```sql
create or replace function delete_storage_object(bucket text, object text, out status int, out content text)
returns record
language 'plpgsql'
security definer
as $$
declare
  project_url text := '<YOURPROJECTURL>';
  service_role_key text := '<YOURSERVICEROLEKEY>'; --  full access needed
  url text := project_url||'/storage/v1/object/'||bucket||'/'||object;
begin
  select
      into status, content
           result.status::int, result.content::text
      FROM extensions.http((
    'DELETE',
    url,
    ARRAY[extensions.http_header('authorization','Bearer '||service_role_key)],
    NULL,
    NULL)::extensions.http_request) as result;
end;
$$;
```

If possible, I would also like to know what I should put for the project url for a local project created using ``supabase/cli``, that will have to be push later to ``production``.

Thanks in advance

**Comment by BragdonD** [comment]
So I find an answer with [Vault](https://supabase.com/docs/guides/database/vault#viewing-secrets).

But I am now struggling to understand how to test the whole table with pgTap...

**Comment by davidroeca** [comment]
Stumbled across this as well. There is an ongoing discussion here related to setting environment variables: https://github.com/orgs/supabase/discussions/12813 -- that should solve it.
